webapp/src/main/java/com/box/l10n/mojito/rest/textunit/StringToDateTimeConverter.java

```diff
@@ -1,33 +1,38 @@
 package com.box.l10n.mojito.rest.textunit;
 
-import org.joda.time.DateTime;
+import java.time.*;
+import java.time.format.DateTimeParseException;
+import org.apache.commons.lang3.StringUtils;
 import org.springframework.core.convert.converter.Converter;
 import org.springframework.stereotype.Component;
 
 /**
- * Converts a {@link String} into {@link DateTime}. The string can either be a number of
- * milliseconds from 1970-01-01T00:00:00Z or any format recognized by {@link
- * DateTime#DateTime(java.lang.Object) }
+ * Converts a {@link String} into {@link ZonedDateTime}. The string can either be a number of
+ * milliseconds from 1970-01-01T00:00:00Z or a string representing a date following the ISO format,
+ * such as '2011-12-03T10:15:30'.
  *
  * @author jeanaurambault
  */
 @Component
-public class StringToDateTimeConverter implements Converter<String, DateTime> {
+public class StringToDateTimeConverter implements Converter<String, ZonedDateTime> {
 
   @Override
-  public DateTime convert(String source) {
+  public ZonedDateTime convert(String source) {
 
-    DateTime converted = null;
-
-    if (source != null) {
-      Object instant;
+    ZonedDateTime converted = null;
 
+    if (StringUtils.isNotEmpty(source)) {
       try {
-        instant = Long.parseLong(source);
+        Instant instant = Instant.ofEpochMilli(Long.parseLong(source));
```

Comment:
why do we need this construct now?

It is surprising that there is no drop-in replacement for `new DateTime(instant)` which seems to be a very common API. 

We should probably have utility class as drop in replacement for all the places we change this. Then if there is a bug that will be easy to find all the usages/regressions